rest-servlet/src/main/java/org/hawkular/inventory/rest/RestResourceTypes.java

```diff
@@ -112,8 +113,9 @@ public ResourceType get(@PathParam("tenantId") String tenantId,
     public Set<Resource> getResources(@PathParam("tenantId") String tenantId,
                                       @PathParam("resourceTypeId") String resourceTypeId) {
 
-        return inventory.tenants().get(tenantId).resourceTypes().get(resourceTypeId)
-                .resources().getAll().entities();
+        ResourceTypes.Single single = inventory.tenants().get(tenantId).resourceTypes().get(resourceTypeId);
```

Comment:
agreed, but I'd differentiate the java api and rest api. I think it's perfectly fine to return the empty set if the filters in the method chain (pretty arbitrary and general approach) doesn't match any entities. 

On the other hand, IMO, we can't do the same for REST endpoints that should return "all resources of given resource type". Returning HTTP 200 with empty JSON array for non-existent resource types is just wrong. (currently what is happening is this: GET http://foo/bar/baz = 404, but http://foo/bar/baz/mazz = 200) Our rest api is just coarse grained, we don't allow for any combination of filters as we do for the Java API. I feel the difference here.